File: Car-Booking-App/src/app/models/booking.ts
import { Car } from "./car";

export class Booking {
    constructor(
      public id?: number,
      public bookingNo?: string,
      public customerName?: string,
      public carId?: number,
      public rentalRate?: number,
      public startDate?: Date,
      public endDate?: Date,
      public car: Car = new Car
    ) {}
  }

File: CarBookingAPI/CarBookingAPI/Models/Booking.cs
namespace CarBookingAPI.Models
{
    public class Booking
    {
        public int Id { get; set; }
        public string BookingNo { get; set; }
        public string CustomerName { get; set; }
        public int CarId { get; set; }
        public decimal RentalRate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public Car Car { get; set; }
    }
}
